tests/preprocessing/test_positionfixes.py

```diff
@@ -228,9 +237,31 @@ def test_temporal(self):
 class TestGenerate_triplegs:
     """Tests for generate_triplegs() method."""
 
-    def test_invalid_isolates(self, isolated_positionfixes):
+    def test_noncontinuous_unordered_index(self, example_positionfixes_isolated):
+        """The unordered and noncontinuous index of pfs shall not affect the generate_triplegs() result."""
+        pfs = example_positionfixes_isolated
+
+        # regenerate noncontinuous index
+        pfs.index = range(0, pfs.shape[0] * 2, 2)
+        pfs.index.name = "id"
+        # unorder index
+        pfs = pfs.sample(frac=1)
+
+        # a warning shall raise due to the duplicated positionfixes
+        warn_string = "The positionfixes with ids .* lead to invalid tripleg geometries."
+        with pytest.warns(UserWarning, match=warn_string):
+            pfs, tpls = pfs.as_positionfixes.generate_triplegs()
+
+        # the index shall be reordered
+        assert np.all(pfs.index == range(0, pfs.shape[0] * 2, 2))
+
+        # only user 1 has generated a tripleg.
+        # user 0 and 2 tripleg has invalid geometry (two identical points and only one point respectively)
+        assert (tpls["user_id"].unique()[0] == 1) and (len(tpls) == 1)
```

Comment:
I think `(tpls["user_id"].unique()[0] == 1)` is quite confusing. How about `(tpls["user_id"].iloc[0] == 1)`

Reply:
the former implies there is only one `user_id` whereas the latter only takes the first record. It is true they are the same under the condition `(len(tpls) == 1)`.

How about `(tpls["user_id"].unique() == [1])`?

Reply:
`tpls["user_id"].unique() == [1]` works for me.